tensorflow/core/grappler/optimizers/auto_mixed_precision.cc

```diff
@@ -1202,20 +1204,24 @@ Status AutoMixedPrecisionImpl::Optimize() {
   TF_RETURN_IF_ERROR(node_type_map_.Init(*graph_));
 
   DataStructureOpsMap object_clients_map;
-  VLOG(2) << "Identifying Stack*V2 nodes";
+
+  // TODO(benbarsdell): Work out how to handle "TensorListPushBackBatch".
+  VLOG(2) << "Identifying TensorList* nodes";
   TF_RETURN_IF_ERROR(AddDataStructureOpsToMap(
-      "StackV2", TypeAttrId("elem_type"), {{"StackPushV2", TypeAttrId("T")}},
-      {{"StackPopV2", TypeAttrId("elem_type")}}, &object_clients_map));
-  VLOG(2) << "Identifying TensorArray*V3 nodes";
-  TF_RETURN_IF_ERROR(
-      AddDataStructureOpsToMap("TensorArrayV3", TypeAttrId("dtype"),
-                               {{"TensorArrayWriteV3", TypeAttrId("T")},
-                                {"TensorArrayScatterV3", TypeAttrId("T")},
-                                {"TensorArraySplitV3", TypeAttrId("T")}},
-                               {{"TensorArrayReadV3", TypeAttrId("dtype")},
-                                {"TensorArrayGatherV3", TypeAttrId("dtype")},
-                                {"TensorArrayConcatV3", TypeAttrId("dtype")}},
-                               &object_clients_map));
+      {"EmptyTensorList", "TensorListSplit", "TensorListFromTensor",
+       "TensorListReserve", "TensorListScatter", "TensorListScatterV2",
+       "TensorListConcatLists"},
```

Comment:
I don't think TensorListConcatLists is handled correctly. You also have to change the dtypes of the input lists.

Maybe give up trying to handle changing the dtype for that tensor list if you encounter a TensorListConcatLists op. Or add a TODO.

Reply:
Thanks for pointing this out. I've worked around this along with TensorListPushBackBatch.